packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -285,32 +307,34 @@ export async function runNpmInstall(
   spawnOpts?: SpawnOptions,
   meta?: Meta
 ) {
-  if (meta && meta.isDev) {
+  if (meta?.isDev) {
     debug('Skipping dependency installation because dev mode is enabled');
     return;
   }
 
   assert(path.isAbsolute(destPath));
   debug(`Installing to ${destPath}`);
 
-  const { cliType } = await scanParentDirs(destPath);
+  const { cliType, lockfileVersion } = await scanParentDirs(destPath);
   const opts: SpawnOptionsExtended = { cwd: destPath, ...spawnOpts };
   const env = opts.env ? { ...opts.env } : { ...process.env };
   delete env.NODE_ENV;
   opts.env = env;
 
-  let command: 'npm' | 'yarn';
   let commandArgs: string[];
 
   if (cliType === 'npm') {
     opts.prettyCommand = 'npm install';
-    command = 'npm';
     commandArgs = args
       .filter(a => a !== '--prefer-offline')
       .concat(['install', '--no-audit', '--unsafe-perm']);
+
+    if (typeof lockfileVersion === 'number' && lockfileVersion >= 2) {
```

Comment:
```suggestion
    if (typeof lockfileVersion === 'number' && lockfileVersion >= 3) {
```

Shouldn't this be version 3 instead since 2 is backwards compatible?

https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion

Reply:
I guess npm 6 only creates version 1 so we can use npm7 for version 2 and newer